Able to complete the training after these changes

Aryal007 opened the pull request:
**ml_classifier** 
- hyperparameter search for best parameters in MLP and added those

**frame.py** 
- fixed output dir
- made loss a class attribute
- changed function name loss -> calc_loss
- changed permute in y value (Wasn't able to run without fixing this)
- infer function returns _(Batch * Channels * Height * Weight)_
- calculate metrics was calculating twice, fixed

**mask.py**
- changed path object to string for concatenation
- changed string to rasterio.crs.CRS objects for comparison
- clipped glaciers to borders.shp (This is needed. I will explain this when you have time)

**post_process.py**
- added default root

**post_process_funs.py**
- create file if not exists
- filter by adding img_mean

**slice.py**
- added img_mean inn json file
- changed soon filename for consistency across files

**train.py**
- batch wise loss visualization using training
- changed path to save_directory

**unet.py**
- changed to 16 channels on first conv_layer

krisrs1128 commented on src/slice.py:
```diff
@@ -115,5 +117,5 @@ def wrapper(row):
     metadata = para(delayed(wrapper)(k) for k in range(len(paths)))
     metadata = pd.concat(metadata, axis=0)
-    out_path = Path(args.output_dir, f"slices_{args.start_line}-{args.end_line}.geojson")
+    out_path = Path(args.output_dir, f"slices_subset.geojson")
```
Won't this overwrite different slices, if you run it in parallel?

krisrs1128 commented on src/frame.py:
```diff
@@ -4,36 +4,39 @@
 import src.metrics
 import src.models.unet
 import torch
+# from torch import optim
```
I'd prefer if you just remove, rather than comment out. We can always add back if we want later.

krisrs1128 commented on src/mask.py:
```diff
@@ -36,14 +37,27 @@ def generate_masks(img_paths, shps_paths, output_base="mask",
+            shps.append(gdf)
+
+        if rasterio.crs.CRS(img.meta["crs"].data) != (rasterio.crs.CRS(shps[0].crs) == rasterio.crs.CRS(shps[1].crs)):
+            print("\nImageCRS: ",rasterio.crs.CRS(img.meta["crs"].data)) 
```
This is smart.

krisrs1128 commented on src/mask.py:
```diff
@@ -36,14 +37,27 @@ def generate_masks(img_paths, shps_paths, output_base="mask",
         print(f"working on image {k} / {len(img_paths)}")
         img, shps = rasterio.open(img_paths[k]), []
+        
+        # print(shps_paths)
```
Same as before -- there are a few points in here where things are commented out, where I'd rather they just be removed. As a general rule, I try to make code as  compact as possible (while maintaining clarity).

krisrs1128 commented on src/postprocess_funs.py:
```diff
@@ -47,7 +48,7 @@ def reshuffle(split_ids, output_dir="output/", n_cpu=3):
             cur_locs = {}
             for im_type in ["img", "mask"]:
-                print(f"shuffling image {i} - {im_type}")
+                # print(f"shuffling image {i} - {im_type}")
```
same as before, if we don't need something, just delete

krisrs1128 commented on src/train.py:
```diff
@@ -19,65 +19,73 @@
 
 ## Train Loop
-epochs=10
+epochs=1000
```
shouldn't this be read in from one of the configuration files?

Aryal007 commented:
**frame.py** 
- valid for model_opts.name "Unet" and "UnetDropout"
- fixed calculate metrics 

**train.py** 
- added training prediction/validation prediction to tensorboard
- saving input configurations as json in tensorboard
- tensorboard saves train/validation loss and metrics
- prettier commandline train runs
- folder structure consistency with previous steps in training pipeline, change the default value of --path in argument parser after pulling.

**unet_dropout.py**
- experimentation capability with spatial dropout added
